src/structs.jl

```diff
@@ -229,6 +236,46 @@ read(::ArrayType, buf, pos, len, b, ::Type{Tuple}, ::Type{eT}; kw...) where {eT}
     invalid(error, buf, pos, T)
 end
 
+@inline readjsonlines(buf, pos, len, b, ::Type{T}; kw...) where {T} = readjsonlines(buf, pos, len, b, T, Base.IteratorEltype(T) == Base.HasEltype() ? eltype(T) : Any; kw...)
+@inline readjsonlines(buf, pos, len, b, ::Type{T}, ::Type{eT}; kw...) where {T, eT} = readjsonlinesarray(buf, pos, len, b, T, eT; kw...)
+
+@inline function readjsonlinesarray(buf, pos, len, b, ::Type{T}, ::Type{eT}; kw...) where {T, eT}
+    vals = Vector{eT}(undef, 0)
+    while true
+        # positioned at start of value
+        @wh_done  # remove spaces and tabs, jump to done on eof
+        pos, y = read(StructType(eT), buf, pos, len, b, eT; kw...)
+        push!(vals, y)
+        if pos > len
+            @goto done
+        end
+        b = getbyte(buf, pos)
+        @wh_done  # remove spaces and tabs, jump to done on eof
+        if b != UInt8('\n') && b != UInt8('\r')
+            error = ExpectedComma
```

Comment:
Maybe we should add an `ExpectedNewline`?

Reply:
Added, and changed `ExpectedComma` to `ExpectedNewline` here and in `read.jl`.